_resources/port1.0/group/haskell_cabal-1.0.tcl

```diff
@@ -133,7 +138,7 @@ default haskell_cabal.build_dir     {${workpath}/dist}
 # use to install prebuilt binaries for bootstrapping
 default haskell_cabal.use_prebuilt  {no}
 
-post-patch {
+post-extract {
```

Comment:
Using post-extract for this has exactly the same problems as post-patch, as indeed would putting it inside any single phase. Any time the phase is already marked as complete in the statefile, it will be skipped and this code won't run. If these variables can't be set up at the global level, then you need to ensure that they are set up in every phase that might use them (preferably by calling a common proc that sets them up only if it hasn't already been done.)

Reply:
It does not appear that MacPorts provides a method to modify paths or environment variables globally. Is there such a Portfile method that avoids the issue you describe?

Reply:
Unfortunately, if you change `::env` it is changed for the entire process, not just the subinterpreter running your Portfile. So any PATH modification has to be done in a temporary way like `command_exec` does it, changing it before doing the thing that needs it and then restoring it to its original value afterwards.

Since you're already defaulting the env option for each phase to `${haskell_cabal.env}`, perhaps lazily evaluating that is all that's needed. Something like this:
```tcl
proc haskell_cabal.get_env {} {
    global haskell_cabal.cabal_root haskell_cabal.use_prebuilt
    set myenv [list CABAL_CONFIG=${haskell_cabal.cabal_root}/config]
    if {[tbool haskell_cabal.use_prebuilt]} {
        lappend myenv \
                    GHC=${haskell_cabal.cabal_root}/bin/ghc \
                    PATH=${haskell_cabal.cabal_root}/bin:$::env(PATH)
    }
    return $myenv
}
default haskell_cabal.env \
        {[haskell_cabal.get_env]}
```